module/profiler/profiler.go

```diff
@@ -71,33 +83,53 @@ func (p *AutoProfiler) Enabled() bool {
 	return p.enabled.Load()
 }
 
-func (p *AutoProfiler) Ready() <-chan struct{} {
-	delay := time.Duration(float64(p.interval) * rand.Float64())
```

Comment:
i've removed the `LaunchPeriodically` because I wanted to not only have the the periodic trigger but also a manual one.  Since we now doing the the scheduling ourselves it is easier to randomize the interval itself instead of introducing the delay on the initial tick of the ticker. (I can't just wait on `time.After` since we want to be able to receive manual trigger during that time.)